apache-kafka-3/src/test/java/com/baeldung/kafka/KafaProducerConsumerAckOptsLiveTest.java

```diff
@@ -0,0 +1,183 @@
+
+        producerProperties.put(ProducerConfig.DELIVERY_TIMEOUT_MS_CONFIG, 10000);
+        producerProperties.put(ProducerConfig.REQUEST_TIMEOUT_MS_CONFIG, 300);
+
+        producerProperties.put(ProducerConfig.LINGER_MS_CONFIG, 10);
+
+        producerProperties.put(ProducerConfig.BUFFER_MEMORY_CONFIG, 33554432);
+        producerProperties.put(ProducerConfig.MAX_BLOCK_MS_CONFIG, 5000);
+
+        return producerProperties;
+    }
+
+    static Properties getConsumerProperties() {
+        Properties consumerProperties = new Properties();
+        consumerProperties.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, KAFKA_CONTAINER.getBootstrapServers());
+        consumerProperties.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());
+        consumerProperties.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());
+        consumerProperties.put(ConsumerConfig.GROUP_ID_CONFIG, CONSUMER_GROUP_ID);
+
+        return consumerProperties;
+    }
+
+    @AfterAll
+    static void destroy() {
+        KAFKA_CONTAINER.stop();
+    }
+
+    @Test
+    @Order(1)
+    void givenProducerAck0_whenProducerSendsRecord_thenDoesNotReturnOffset() throws ExecutionException, InterruptedException {
+        ProducerRecord<String, String> record = new ProducerRecord<>(TOPIC, 0, MESSAGE_KEY, TEST_MESSAGE + "_0");
+        for (int i = 0; i < 50; i++) {
+            RecordMetadata metadata = producerack0.send(record)
+                .get();
+            assertEquals(INVALID_OFFSET, metadata.offset());
+        }
+    }
+
+    @Test
+    @Order(2)
+    void givenProducerAck1_whenProducerSendsRecord_thenReturnsValidOffset() throws ExecutionException, InterruptedException {
+        ProducerRecord<String, String> record = new ProducerRecord<>(TOPIC, 1, MESSAGE_KEY, TEST_MESSAGE + "_1");
+        for (int i = 0; i < 50; i++) {
+            RecordMetadata metadata = producerack1.send(record)
+                .get();
+            assertNotEquals(INVALID_OFFSET, metadata.offset());
+        }
+    }
+
+    @Test
+    @Order(3)
+    void givenProducerAckAll_whenProducerSendsRecord_theReturnsValidOffset() throws ExecutionException, InterruptedException {
+        ProducerRecord<String, String> record = new ProducerRecord<>(TOPIC, 2, MESSAGE_KEY, TEST_MESSAGE + "_ALL");
+        for (int i = 0; i < 50; i++) {
+            RecordMetadata metadata = producerackAll.send(record)
+                .get();
+            assertNotEquals(INVALID_OFFSET, metadata.offset());
+        }
+    }
+
+    @Test
+    @Order(4)
+    void whenSeekingKafkaResetConfigLatest_consumerOffsetSetToLatestRecordOffset() {
+        Properties consumerProperties = getConsumerProperties();
+        consumerProperties.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, false);
+        consumerProperties.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "latest");
+        long expected_start_offset = 50;
+        long actual_start_offset = -1;
```

Comment:
We follow https://google.github.io/styleguide/javaguide.html#s5.2.7-local-variable-names for naming